build_resolvers/test/resolver_test.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:build_test/build_test.dart';
+import 'package:test/test.dart';
+
+import 'package:build_resolvers/build_resolvers.dart';
+import 'package:build_resolvers/src/resolver.dart';
+
+void main() {
+  final entryPoint = new AssetId('a', 'web/main.dart');
+
+  group('Resolver', () {
+    test('should handle initial files', () {
+      return resolveSources({
+        'a|web/main.dart': ' main() {}',
+      }, (resolver) async {
+        var lib = await resolver.libraryFor(entryPoint);
+        expect(lib, isNotNull);
+      }, resolvers: new AnalyzerResolvers());
+    });
+
+    test('should follow imports', () {
+      return resolveSources({
+        'a|web/main.dart': '''
+              import 'a.dart';
+
+              main() {
+              } ''',
+        'a|web/a.dart': '''
+              library a;
+              ''',
+      }, (resolver) async {
+        var lib = await resolver.libraryFor(entryPoint);
+        expect(lib.importedLibraries.length, 2);
+        var libA = lib.importedLibraries.where((l) => l.name == 'a').single;
+        expect(libA.getType('Foo'), isNull);
+      }, resolvers: new AnalyzerResolvers());
+    });
+
+    test('should follow package imports', () {
+      return resolveSources({
+        'a|web/main.dart': '''
+              import 'package:b/b.dart';
+
+              main() {
+              } ''',
+        'b|lib/b.dart': '''
+              library b;
+              ''',
+      }, (resolver) async {
+        var lib = await resolver.libraryFor(entryPoint);
+        expect(lib.importedLibraries.length, 2);
+        var libB = lib.importedLibraries.where((l) => l.name == 'b').single;
+        expect(libB.getType('Foo'), isNull);
+      }, resolvers: new AnalyzerResolvers());
+    });
+
+    test('handles missing files', () {
+      return resolveSources({
+        'a|web/main.dart': '''
+              import 'package:b/missing.dart';
+
+              main() {
+              } ''',
+      }, (resolver) async {
+        var lib = await resolver.libraryFor(entryPoint);
+        expect(lib.importedLibraries.length, 2);
+      }, resolvers: new AnalyzerResolvers());
+    });
+
+    //test('should list all libraries', () {
```

Comment:
is this still failing or just accidentally commented out?

Reply:
Just a bad rebase after we got the tests working.